File: test8/src/main/java/com/example/test8/DBHelper.java
package com.example.test8;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.DatabaseErrorHandler;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.support.annotation.Nullable;

public class DBHelper extends SQLiteOpenHelper {
    private String TABLE = "song";

    public DBHelper(@Nullable Context context) {
        super(context, "song_db", null, 1);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL("CREATE TABLE if not exists " + TABLE + "(_id integer PRIMARY KEY AUTOINCREMENT, genre text, title text, content text)");
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

    }

    public void addStudent(String genre, String title, String content){
        ContentValues newData = new ContentValues();
        newData.put("genre",genre);
        newData.put("title",title);
        newData.put("content", content);
        getWritableDatabase().insert(TABLE, null, newData);
    }

    public Cursor getStudent(String title){
        return getReadableDatabase().query(TABLE, new String[]{"_id","genre","title","content"},"title=?",new String[]{title}, null,null, null,null);
    }
}

File: app/src/main/java/com/example/sqliteexample/DBHelper.kt
package com.example.sqliteexample

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.util.Log

class DbHelper(context: Context) : SQLiteOpenHelper(context, "school.db", null, 1) {
    val TAG = "####"
    val TABLE = "student"

    companion object {
        val _id: String = "_id"
        val genre: String = "genre"
        val title: String = "title"
        val content: String = "content"
    }

    val DATABASE_CREATE =
        "CREATE TABLE if not exists " + TABLE + " (" +
                "${_id} integer PRIMARY KEY AUTOINCREMENT," +
                "${genre} text," +
                "${title} text,"+
                "${content} text"+
                ")"

    fun addStudent(_id:String, _name:String, _phone: String, _depart:String) {
        val values = ContentValues()
        values.put(num, _num)
        values.put(name, _name)
        values.put(phone, _phone)
        values.put(depart, _depart)
        writableDatabase.insert(TABLE, null, values);
    }

    fun getStudent() : Cursor {
        return readableDatabase.query(TABLE, arrayOf(num, name, phone, depart), null, null, null, null, null);
    }

    override fun onCreate(db: SQLiteDatabase) {
        Log.d(TAG, "Creating: " + DATABASE_CREATE);
        db.execSQL(DATABASE_CREATE)
    }

    override fun onUpgrade(p0: SQLiteDatabase, p1: Int, p2: Int) {
        val dropSQL = "drop table student"
    }

}
